CONTRIBUTING.md

```diff
@@ -0,0 +1,900 @@
+		if (f(list[i], i) === false) {
+			// If the function explicitly returns `false`, immediately stop
+			// iteration.
+			break
+		}
+	}
+	// Return the list, for convenience.
+	return list
+}
+```
+
+### Initial whitespace
+
+Start your comments with a space. It's more readable.
+
+```js
+// Good
+// This is a comment.
+
+// Bad
+//This is a comment.
+```
+
+### Grammar and spelling
+
+Use proper grammar and spelling in your comments. This shouldn't need explanation.
+
+## Whitespace
+
+Use vertical whitespace and indentation to your advantage. It helps with readability, and minifiers are astonishingly great at removing this. :stuck_out_tongue_winking_eye:
+
+```js
+// Good
+function iterate() {
+	var index = 0
+	var list = []
+	while (hasNext()) {
+		list.push(next(index))
+		index++
+	}
+
+	for (var i = 0; i < list.length; i++) {
+		read(list[i])
+	}
+}
+
+// Bad
+function iterate() {
+	var index = 0
+	var list = []
+	while (hasNext()) list.push(next(index++))
+	for (var i = 0; i < list.length; i++) read(list[i])
+}
+
+// Even worse
+function iterate() {
+	var list = []
+	for (var index = 0; hasNext(); list.push(next(index++))) {}
+	for (var i = 0; i < list.length; i++) read(list[i])
+}
+```
+
+### Trailing whitespace
+
+Please don't leave trailing whitespace anywhere. It makes diffs harder to read.
+
+Blank lines should have no indentation or spaces. It should be empty other than a line ending.
+
+All files should end with a trailing line break. Some editors add this on their own, so it will result in unnecessary lines in later diffs if you don't.
+
+### Indentation and vertical whitespace
+
+Indent with hard tabs. Each one counts for 4 spaces.
+
+Never mix tabs and spaces. Don't use smart tabs.
+
+### Excessive whitespace
+
+Keep whitespace within reason. Limit vertical whitespace to no more than 2 consecutive blank lines, and don't start or end blocks with plain whitespace. Don't use more than one character of horizontal whitespace beyond indentation.
+
+```js
+// Bad
+function iterate() {
+	var index = 0
+	var list = []
+
+	while (hasNext()) {
+		list.push(next(index))
+		index++
+	}
+
+	// Too many empty lines
+
+
+
+	for (var i = 0; i < list.length; i++) {
+		read(list[i])
+	}
+}
+
+// Also bad
+var a  =  2
+//   ^^ ^^
+```
+
+### Control keywords
+
+Always surround control keywords (e.g. `if`, `else`, `for`) with whitespace.
+
+## Operators
+
+### Binary operators
+
+Always surround binary keyword operators (e.g. `in`, `instanceof`) with whitespace.
+
+Always surround any other binary operator with whitespace, including assignment operators. Add line breaks after the operator, not before.
+
+```js
+// Good
+var a = 1 + 2
+a = 3
+a += 4
+
+var a = 1 + 2 + 3 + 4 + 5 +
+	6 + 7 + 8 + 9 + 10
+
+// Bad
+var a = 1+2
+var a = 1 +2
+var a = 1+ 2
+var a=1+2
+var a=1 + 2
+a=3
+a+=4
+
+var a = 1 + 2 + 3 + 4 + 5
+	+ 6 + 7 + 8 + 9 + 10
+```
+
+As an exception, casting to a 32-bit integer (i.e. `x|0`) doesn't require whitespace surrounding it.
```

Comment:
I've made it more apparent that this is more of a forward-looking rule, and that it's not currently used in Mithril itself. It's mostly for performance-sensitive code, where you want to ensure the engine knows it's dealing with exclusively integers, which are far faster than floats (it's not easy to tell the difference between an integer and a float).